tools/crm_attribute.c

```diff
@@ -503,9 +503,20 @@ main(int argc, char **argv)
         if (target != NULL) {
             g_free(options.dest_uname);
             options.dest_uname = g_strdup(target);
+        } else if (getenv("CIB_file") != NULL) {
+            char *hostname = pcmk_hostname();
+
+            g_free(options.dest_uname);
+
+            /* This silliness is so that dest_uname is always a glib-managed
+             * string so we know how to free it later.  pcmk_hostname returns
+             * a newly allocated string via strdup.
+             */
```

Comment:
pcmk_hostname() is a one-liner, you could just use uname() directly if you want to avoid the double memory (not that it really matters)